libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -145,9 +162,32 @@ void AC_PrecLand::init()
     }
 }
 
+// create inertial buffer
+void AC_PrecLand::check_inertial_buffer()
+{
+    // if lag parameter is not within bounds, return without doing anything
+    if (_lag_ms <= 0 || _lag_ms > 250) {
+        return;
+    }
+
+    // calculate inertial buffer size from delay length (loop runs at 400hz)
+    uint16_t inertial_buffer_size = (uint16_t)roundf(_lag_ms / (1000/400.0f));
+
+    // create buffer if it doesn't exist or if latency parameter has changed
+    if (_inertial_history == nullptr || inertial_buffer_size != _inertial_history->size()) {
```

Comment:
You can get in here if the allocated buffer size doesn't match the desired size, but you never free the old object which can leak memory.

Reply:
check_initial_buffer() was being called from update so it could, as you say, lead to a memory leak but I've "fixed" that by only calling it from init. I did debate whether I should remove the size check .. it doesn't do anything anymore but I left it in anyway.. maybe I'll remove it..

Reply:
I'm primarily calling it out since it's in a function called `check_inertial_buffer` which I would have expected to be fair to call in other places. If we are only doing this in init we should probably just have the actual allocation code there and can avoid the nullpointer check as well.

Reply:
OK, i'll rename it to init_inertial_buffer() and remove the extra check.  Txs very much for the review!

Reply:
.. the function should probably be private as well..

Reply:
Ah, combining it into init would have been good...  I'll leave that to another day though 'cuz I don't feel like retesting yet again.

Reply:
Is there any way the memory allocation/destruction can be fixed so this can be called from update?  When you're first tuning a system to work by trial and error (if you don't have an accurate measurement of your sensor lag), having to reboot the firmware after every change is a bit of a pain.  I used this quite a bit during testing to change the buffer size during a quick land and then take off again for another go.
Not a big deal, but would be nice from a user perspective if it could work.

Reply:
@fnoop, yes, I was thinking that's probably why you'd done it that way.  The issue was we would end up with a (relatively small) memory leak every time the parameter was changed.  We could add a delete which would probably be fine although as a general rule we try to stay away from doing too many new and delete calls especially when the vehicle is flying.  A better solution would probably be to try and access a different part of the ring buffer depending upon the LAG parameter.  I.e. we allocate a bigger buffer, but then just access somewhere earlier in it.

Thanks very much for reviewing.